sources/AppBundle/Controller/Admin/Event/EventAction.php

```diff
@@ -0,0 +1,120 @@
+                $this->flashBag->add('error', 'Évènement non trouvé');
+                return new RedirectResponse($this->urlGenerator->generate('admin_event_list'));
+            }
+        }
+
+        $form = $this->formFactory->create(EventType::class, $event);
+
+        if ($event->getId() && $couponsImploded = $this->couponRepository->couponsListForEventImploded($event)) {
+            $form->get('coupons')->setData($couponsImploded);
+        }
+
+        $form->handleRequest($request);
+        if ($form->isSubmitted() && $form->isValid()) {
+            $this->moveSponsorFile($event, $form, 'fr');
+            $this->moveSponsorFile($event, $form, 'en');
+            $this->moveRegistrationEmailFile($event, $form);
+
+            $this->eventRepository->save($event);
+
+            if ($form->get('coupons')->getData()) {
+                $eventCoupons = explode(',', $form->get('coupons')->getData());
+                $this->couponRepository->changeCouponForEvent($event, $eventCoupons);
+            }
+
+            $this->flashBag->add('notice', 'Évènement ' . ($id ? 'modifié' : 'ajouté'));
+            return new RedirectResponse($this->urlGenerator->generate('admin_event_edit', ['id' => $event->getId()]));
+        }
+
+        $sponsorFilePathFR = Event::hasSponsorFile($event->getPath(), 'fr') ? Event::getSponsorFilePublicPath($event->getPath(), 'fr') : null;
+        $sponsorFilePathEN = Event::hasSponsorFile($event->getPath(), 'en') ? Event::getSponsorFilePublicPath($event->getPath(), 'en') : null;
+        $registrationEmailFilePath = Event::hasInscriptionAttachment($event->getPath()) ? Event::getInscriptionAttachmentPublicPath($event->getPath()) : null;
+
+        return new Response($this->twig->render('admin/event/' . ($id ? 'edit' : 'add') . '.html.twig', [
+            'form' => $form->createView(),
+            'event' => $event,
+            'sponsor_file_path_fr' => $sponsorFilePathFR,
+            'sponsor_file_path_en' => $sponsorFilePathEN,
+            'registration_email_file_path' => $registrationEmailFilePath,
+        ]));
+    }
+
+    private function moveSponsorFile(Event $event, FormInterface $form, string $language): void
+    {
+        /** @var ?UploadedFile $uploadedFile */
+        $uploadedFile = $form->get('sponsor_file_' . $language)->getData();
+        if ($uploadedFile) {
+            $dir = Event::getSponsorFileDir();
+            if (!is_dir($dir) && !mkdir($dir) && !is_dir($dir)) {
+                throw new \RuntimeException(sprintf('Directory "%s" was not created', $dir));
+            }
+            $filename = Event::getSponsorFilePath($event->getPath(), $language);
+            $uploadedFile->move($dir, basename($filename));
+        }
+    }
+
+    private function moveRegistrationEmailFile(Event $event, FormInterface $form): void
+    {
+        /** @var ?UploadedFile $uploadedFile */
+        $uploadedFile = $form->get('registration_email_file')->getData();
+        if ($uploadedFile) {
```

Comment:
Petit proposition pour rendre le code plus safe :

```suggestion
        $uploadedFile = $form->get('registration_email_file')->getData();
        if ($uploadedFile instanceof UploadedFile) {
```